### ISSUE TITLE
Testing file names format

### ISSUE THREAD

**Issue by GaryJones** [squizlabs/PHP_CodeSniffer#1078]
For the WordPress standard, we've got a sniff that checks for underscores in the filename, as generally, they are disallowed. However, there are [exceptions](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/470). While creating the regex pattern to detect those exceptions isn't tricky, coming up with PHPCS unit tests is.

Since PHPCS unit tests files apparently need to be named in a particular way to be found and run automatically, we can't use them to test file name variations.

Would be possible to have sniff files register additional test files that they want to use and have the test runner pick them up?

Alternatively, having the ability to override the `$phpcs->fileName` with some virtual file names may also work for our particular use case.

**Comment by gsherwood** [comment]
I ended up needing this to support PR #1069, so it's done now.

There is a committed example here: https://github.com/squizlabs/PHP_CodeSniffer/blob/master/CodeSniffer/Standards/Generic/Tests/PHP/DisallowShortOpenTagUnitTest.php#L42

The method linked above looks like this:
```php
    /**
     * Get a list of all test files to check.
     *
     * @param string $testFileBase The base path that the unit tests files will have.
     *
     * @return string[]
     */
    protected function getTestFiles($testFileBase)
    {
        $testFiles = array($testFileBase.'1.inc');

        $option = (boolean) ini_get('short_open_tag');
        if ($option === true || defined('HHVM_VERSION') === true) {
            $testFiles[] = $testFileBase.'2.inc';
        }

        return $testFiles;

    }//end getTestFiles()
```

In the `getTestFiles()` method, the `$testFileBase` is the absolute path to the sniffs `...UnitTest.php` file. You can use that to find test files (as I've done above) or use it to find the test directory (`dirname()` it) or just return absolute paths to whatever test files you want.

**Comment by GaryJones** [comment]
Excellent, thank you!
Will this be released in PHPCS 2.8.0?

**Comment by gsherwood** [comment]
> Will this be released in PHPCS 2.7.0?

Yes. Not 100% sure when, but probably a few weeks. Just a few more PRs to get through.
